### DISCUSSION TITLE
Livewire V3 not working with laravel MIX

### DISCUSSION THREAD

**Discussion by iislamtaleb** [livewire/livewire#7848] [1. Help]
I am working on a project and needed to install the Mix plugin alongside Vite. However, I encountered an issue when creating new Livewire components after installing Laravel Mix, Livewire didn't work

**Comment by danie-ramdhani** [comment]
read this (in the bottom): https://livewire.laravel.com/docs/installation#manually-bundling-livewire-and-alpine

**Reply by iislamtaleb** [reply]
Ok, I have upgraded from MIX to Vite, but Im facing the same issue!

![Capture d'écran 2024-01-30 224317](https://github.com/livewire/livewire/assets/148808316/09c1a9b8-2a32-4139-bda1-71487f8552df)
![Capture d'écran 2024-01-30 224344](https://github.com/livewire/livewire/assets/148808316/99dd9308-5ffb-4014-9450-64397569488c)
![Capture d'écran 2024-01-30 224402](https://github.com/livewire/livewire/assets/148808316/0f2ed486-a8bc-45c3-88dd-51b0340d7471)

**Reply by joshhanley** [reply]
@iislamtaleb how are you loading your livewire component? What does the route look like?

Also please don’t paste screenshots of code, because we can’t change them to help you.

Instead paste code in code blocks. When you paste code into github, you need to surround it in triple back ticks and then specify the file type like php or blade.

For example
\```php
\\ PHP code here
\```

\```blade
\\ HTML and blade code here
\```

**Reply by iislamtaleb** [reply]
sorry my bad, I will ask again.


I am working on a project and needed to install the Mix plugin alongside Vite. after that I have removed Mix and Install Vite again and I come to the issue of livewire not working.

for example:
```

 #[Validate('required|string|max:255')]
    public $selectedCategory;
    #[Validate('required|string|max:255')]
    public $selectedDoctor;

    
    public function render()
    {
        return view('livewire.appointment-booking');
    }

    #[Computed()]
    public function categories()
    {
        return Category::all();    
    }

    #[Computed()]
    public function doctors()
    {
        return Doctor::where('category_id',$this->selectedCategory)->get();   
    }
```

```
<div class="col-12 col-sm-6">
  <select class="form-select border-0" style="height: 55px;" name="selectedCategory"     wire:model.live='selectedCategory'>
  <option value="" selected>Select Department</option>
      @foreach ($this->categories as $category)
           <option value="{{ $category->id }}">{{ $category->title }}</option>
      @endforeach
  </select>
  @error('selectedCategory') <span class="text-danger">{{ $message }}</span> @enderror
</div>


<div class="col-12 col-sm-6">
   <select class="form-select border-0" style="height: 55px;" name="selectedDoctor" wire:model.live='selectedDoctor'>
       <option value="" selected>Choose Doctor</option>
       @foreach ($this->doctors as $doctor)
       <option value="{{ $doctor->id }}">{{ $doctor->name }}</option>
       @endforeach
   </select>
   @error('selectedDoctor') <span class="text-danger">{{ $message }}</span> @enderror
</div>
```

The error:

Using $this when not in object context

```
@foreach ($this->categories as $category)

  <option value="{{ $category->id }}">{{ $category->title }}</option>

@endforeach
```
Note: The old livewire files are working fine, but the new ones are not working!

**Reply by joshhanley** [reply]
@iislamtaleb see my questions above:

> how are you loading your livewire component? What does the route look like?

Also screenshots of errors are fine 🙂
